edx_event_bus_redis/internal/tests/test_consumer.py

```diff
@@ -209,14 +219,64 @@ def raise_exception():
         mock_sleep.assert_not_called()
         self.event_consumer.db.close.assert_called_once_with()  # since shutdown was requested, not because of exception
 
+    @patch('edx_event_bus_redis.internal.consumer.set_custom_attribute', autospec=True)
+    @patch('edx_event_bus_redis.internal.consumer.time.sleep', autospec=True)
+    def test_consume_loop_with_no_events(self, mock_sleep, mock_set_custom_attribute):
+        """
+        Check the loop lifecycle if no events are triggered.
+        """
+        # How the _is_fatal_redis_error() mock will behave on each successive call.
+        mock_side_effects = [
+            lambda: None,  # accept and ignore a message
+            # Final "call" just serves to stop the loop
+            self.event_consumer._shut_down  # pylint: disable=protected-access
+        ]
+        mock_read_value = []
+        mock_pending_value = None
+        side_effect_method = '_is_fatal_redis_error'
+
+        with patch.object(
+            self.event_consumer, side_effect_method,
+            side_effect=side_effects(mock_side_effects),
+        ) as mock_method:
+            mock_consumer = MagicMock(
+                **{
+                    'read.return_value': mock_read_value,
+                    'pending.return_value': mock_pending_value,
+                    '__getitem__.return_value': None,
+                },
+                autospec=True
+            )
+            self.event_consumer.db = Mock()
+            self.event_consumer.consumer = mock_consumer
+            self.event_consumer.consume_indefinitely()
+
+        # Check that each of the mocked out methods got called as expected.
+        mock_consumer.pending.assert_called_with(count=1, consumer='test_group_id.c1')
+        mock_consumer.read.assert_called()
+        mock_consumer.ack.assert_not_called()
+        # Check that emit was called the expected number of times
```

Comment:
Copypasta. We're not checking the emit method. I don't think it tells us anything to check the number of calls on the _is_fatal_redis_error.